components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -3512,4 +3519,164 @@ private boolean isTenantDomainNotMatching(String tenantDomain) {
+            File dir = new File("repository/deployment/server//throttlingconfigs");
+            if(!dir.exists()){
+                dir.mkdir();
+            }
+            File file = new File("repository/deployment/server/throttlingconfigs/"+tierName+".xml");
+            fos = new FileOutputStream(file);
+            eligibility_query.addChild(eligibilityQueryText);
+            decision_query.addChild(decisionQueryText);
+            policyTag.addChild(eligibility_query);
+            policyTag.addChild(decision_query);
+            policyTag.build();
+            String policy = policyTag.toString();
+            System.out.println(policyTag.toString());
+            byte[] contentInBytes = policy.getBytes();
+
+            fos.write(contentInBytes);
+            fos.flush();
+            fos.close();
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void jsonWrite(Policy policy){
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        try {
+            File file2 = new File("repository/conf/" + policy.getPolicyName() + ".txt");
+            FileOutputStream fos = null;
+            fos = new FileOutputStream(file2);
+
+            String write = gson.toJson(policy);
+            System.out.println(write);
+            byte[] contentWrite = write.getBytes();
+
+            fos.write(contentWrite);
+            fos.flush();
+            fos.close();
+        }catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public String getCondition(Policy policy) {
+        ArrayList<Condition> conditions = policy.getConditions();
+        String httpVerb = null;
+        String startingIP = null;
+        String endingIP = null;
+        String condition = "";
+        String ifThenElse = "ifThenElse(";
+        String elseCondition = "'elseCondition'";
+        for (int i = 0; i < conditions.size(); i++) {
+            condition += ifThenElse;
+            httpVerb = conditions.get(i).getHttpVerb();
+            startingIP = conditions.get(i).getStartingIP();
+            endingIP = conditions.get(i).getEndingIP();
+
+            if (!httpVerb.isEmpty()) {
+                condition += "verb == '" + httpVerb + "'";
+            }
+            if (!startingIP.isEmpty() && !endingIP.isEmpty()) {
+                if (!httpVerb.isEmpty()) {
+                    condition += " AND ";
+                }
+                String ipRange = "ip>=" + ipToLong(startingIP) + " AND " + ipToLong(endingIP) + ">=ip";
+                condition += ipRange;
+            }
+            condition += ", 'condition" + i + "',";
+        }
+        condition += elseCondition;
+        for (int i = 0; i < conditions.size(); i++) {
+            condition += ")";
+        }
+        return condition;
+    }
+
+    public File getFile(String fileName) throws APIManagementException{
+        File file= new File("repository/deployment/throttlingconfigs/"+fileName+".xml");
```

Comment:
With the next improvement this no need to be a separate method. Let's include it within the method itself.